Tests/Weaver/Realm.Fody.Tests/AnalyticsTests.cs

```diff
@@ -104,19 +104,45 @@ public void Submit_WhenDisabled_PayloadIsEmpty()
             }
         }
 
-        private void ValidateAnalyticsPayload<T>(string featureName, T expectedResult)
+        [Test]
+        public void ValidateHostOS()
         {
-            foreach (var framework in _frameworks.Value)
+            var currentOs = Environment.OSVersion;
+            try
             {
-                var response = WeaveRealm(framework, "DryRun");
-                var payload = BsonSerializer.Deserialize<BsonDocument>(response).AsBsonDocument;
-                Assert.That(payload[featureName].AsString, Is.EqualTo(expectedResult.ToString()), $"Feature {featureName} was not reported as used: {expectedResult} in {framework}");
+                CompileAnalyticsProject();
+                ValidateAnalyticsPayloadAllFrameworks(new[] {
+                    (featureName: Metric.Environment.HostOsType, expectedValue: ConvertToMetricOS(currentOs.Platform)),
+                    (featureName: Metric.Environment.HostOsVersion, expectedValue: currentOs.Version.ToString()) });
+            }
+            catch (Exception e)
+            {
+                Assert.Fail($"{e.Message}");
```

Comment:
Why do we need this? An uncaught exception would fail the test anyway.

Reply:
Because in the past I wanted to given some extra contextual info, which then I stopped doing and forgot to remove the try/catch altogether. Remove it.